app/controllers/app.py

```diff
@@ -1519,3 +1521,61 @@ def post(self, project_id, app_id):
             return dict(status='fail', message='No values found'), 404
 
         return dict(status='success', data=dict(storage_capacity=values, storage_percentage_usage=volume_perc_value)), 200
+
+
+class AppDataSummaryView(Resource):
+
+    @admin_required
+    def post(self):
+        """
+        Aggregate apps per month or year
+        """
+        app_filter_data = request.get_json()
+        filter_schema = AppGraphSchema()
+
+        validated_query_data, errors = filter_schema.load(app_filter_data)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        start = validated_query_data.get('start', '2018-01-01')
+        end = validated_query_data.get('end', datetime.datetime.now())
+        set_by = validated_query_data.get('set_by', 'month')
+        total_apps = len(App.find_all())
+
+        if set_by == 'month':
+            date_list = func.generate_series(
+                start, end, '1 month').alias('month')
+            month = column('month')
+
+            app_data = db.session.query(month, func.count(App.id)).\
+                select_from(date_list).\
+                outerjoin(App, func.date_trunc('month', App.date_created) == month).\
+                group_by(month).\
+                order_by(month).\
+                all()
+
+        else:
+            date_list = func.generate_series(
+                start, end, '1 year').alias('year')
+            year = column('year')
+
+            app_data = db.session.query(year, func.count(App.id)).\
```

Comment:
Hey @MutegekiHenry, great work. But maybe can we have this moved to the Model Class that contains class queries